File: threedollar-api-user-service/src/integration-test/java/com/depromeet/threedollar/api/userservice/SetupUserIntegrationTest.java
package com.depromeet.threedollar.api.userservice;

import org.junit.jupiter.api.BeforeEach;
import org.springframework.beans.factory.annotation.Autowired;

import com.depromeet.threedollar.domain.rds.domain.userservice.user.User;
import com.depromeet.threedollar.domain.rds.domain.userservice.user.UserFixture;
import com.depromeet.threedollar.domain.rds.domain.userservice.user.UserRepository;
import com.depromeet.threedollar.domain.rds.domain.userservice.user.UserSocialType;

public abstract class SetupUserIntegrationTest extends IntegrationTest {

    @Autowired
    protected UserRepository userRepository;

    protected Long userId;

    protected User user;

    @BeforeEach
    void setup() {
        user = userRepository.save(UserFixture.create("social-id", UserSocialType.KAKAO, "디프만"));
        userId = user.getId();
    }

}

File: threedollar-api-admin-service/src/integrationTest/kotlin/com/depromeet/threedollar/api/adminservice/SetupUserIntegrationTest.kt
package com.depromeet.threedollar.api.adminservice

import com.depromeet.threedollar.domain.rds.domain.userservice.user.User
import com.depromeet.threedollar.domain.rds.domain.userservice.user.UserFixture
import com.depromeet.threedollar.domain.rds.domain.userservice.user.UserRepository
import org.junit.jupiter.api.BeforeEach
import org.springframework.beans.factory.annotation.Autowired

internal abstract class SetupUserIntegrationTest : SetupAdminIntegrationTest() {

    @Autowired
    protected lateinit var userRepository: UserRepository

    protected var userId: Long = -1L

    protected lateinit var user: User

    @BeforeEach
    override fun setup() {
        user = userRepository.save(UserFixture.create())
        userId = user.id
    }

}
